### ISSUE TITLE
Memory leak?

### ISSUE THREAD

**Issue by ttuleyb** [neonbjb/tortoise-tts#180]
Hello, I'm working on a UI for tortoise https://github.com/ttuleyb/TortoiseTTS-GUI and I've been wondering how to unload the tortoise models and modules that are eating up a lot of memory while voices aren't being created. I just cannot figure how to do this, so I came here to ask for help thanks.

Btw I'm not talking about the voice cache function, I'm talking about this line of code that increases memory usage quite heavily (I assume its loading the tortoise model)
```py
gen = tts.tts_with_preset(text, voice_samples=reference_clips, preset=preset, k=numOfOutputs)
```
But after the job is complete, the model is never unloaded.

One solution I can think of is to write an external script to connect the API's together and then use the command line to call this script so that when the job is finished, the entire script exits but surely there must be a way to do this within python itself.

**Comment by neonbjb** [comment]
Hey there, cool project!

Are you concerned about CPU memory or GPU memory? Assuming GPU - pytorch (which Tortoise uses) has an [aggressive caching mechanism](https://pytorch.org/docs/stable/notes/cuda.html#memory-management which does not free large memory allocations unless specifically instructed.

**Comment by ttuleyb** [comment]
Hey thank you, that was massively helpful. I've now been able to add a longform option to the UI. 

This project is increadible thanks for making it. Feel free to checkout the project I don't think I'll add much maybe in the future except the ability to directly add reference clips and new people but thats about it.

I respect the fact that you keep the training method/scripts to yourself I actually wonder how long it will take for someone to make something like this. The closest alternative I've found is [Uber Duck](https://uberduck.ai) Which sounds much worse than TortoiseTTS.
